sys/net/application_layer/gcoap/fileserver.c

```diff
@@ -32,14 +32,24 @@
 /** Maximum length of an expressible path, including the trailing 0 character. */
 #define COAPFILESERVER_PATH_MAX (64)
 
+struct requestoptions {
+    struct {
+        bool if_match        :1;
+        bool if_none_match   :1;
+        bool etag            :1;
+        bool block2          :1;
+        bool block1          :1;
+    } exists;
```

Comment:
IIRC there was a problem with flag fields (might be C++ or XCode related, but not sure), could be that I am cargo culting here. However, given that this is at max 8-bit long for now, this should go to the end of the sturct in any case to avoid padding.

Reply:
If there is a problem with bit fields, the CI will show this, right? Then I can of course turn this into flags.
I only know that you should not rely on the order of bits in bit fields.

Reply:
> If there is a problem with bit fields, the CI will show this, right?

Not everything is tested, especially not XCode, as MacOSX support is currently broken for native (and we do not have a MacOSX worker). Not sure if there is a C++ application that uses this module.

Reply:
The only problem I know of with bit fields is that they don't work like one might expect *when using the struct in type punning with a buffer* -- ie. you can't do `union { struct { bool ver: 2; bool mtype: 2; ...} parsed; uint8_t[4] for_network };` but for regular use I think they're fine.

Reply:
`USEMODULE+="gcoap_fileserver gnrc_sock_udp gnrc_ipv6" make BOARD=nucleo-f767zi` compiles

Reply:
for `examples/riot_and_cpp`

Reply:
> The only problem I know of with bit fields is that they don't work like one might expect _when using the struct in type punning with a buffer_ -- ie. you can't do `union { struct { bool ver: 2; bool mtype: 2; ...} parsed; uint8_t[4] for_network };` but for regular use I think they're fine.

Then my cargo cult might stem from that, because I think I got hurt way back when I tried to send them over the wire :D.

Reply:
I consider this discussion resolved then :-).